pkg/storage/storage.go

```diff
@@ -187,13 +190,29 @@ type ChangelogBackend interface {
 	ReadChanges(ctx context.Context, store, objectType string, paginationOptions PaginationOptions, horizonOffset time.Duration) ([]*openfgapb.TupleChange, []byte, error)
 }
 
+// TimeProvider has functionality of Now that allows mocking of current time
+type TimeProvider interface {
+	Now() time.Time
+}
+
+// DefaultTimeProvider allows returning current time as Now()
+type DefaultTimeProvider struct{}
+
+// Now returns the current time
+func (d DefaultTimeProvider) Now() time.Time {
+	return time.Now()
+}
+
 type OpenFGADatastore interface {
 	TupleBackend
 	AuthorizationModelBackend
 	StoresBackend
 	AssertionsBackend
 	ChangelogBackend
 
+	// UpdateCurrentTime allows overriding time.Now functionality
+	UpdateCurrentTime(currentTime TimeProvider)
```

Comment:
```suggestion
        // SetTimeProvider allows overriding time.Now functionality to set timestmaps in writes.
	SetTimeProvider(provider TimeProvider)
```

Reply:
Obsolete as we don't have the time provider anymore